language/diem-vm/src/diem_transaction_executor.rs

```diff
@@ -502,33 +502,46 @@ impl DiemVM {
         {
             return Ok(discard_error_vm_status(e));
         };
-
-        let change_set = match txn.payload() {
-            TransactionPayload::WriteSet(writeset_payload) => {
-                match self.execute_writeset(
-                    remote_cache,
-                    writeset_payload,
-                    Some(txn_data.sender()),
-                    log_context,
-                ) {
-                    Ok(change_set) => change_set,
-                    Err(e) => return e,
+        self.execute_writeset_transaction(
+            remote_cache,
+            match txn.payload() {
+                TransactionPayload::WriteSet(writeset_payload) => writeset_payload,
+                TransactionPayload::Module(_) | TransactionPayload::Script(_) => {
+                    log_context.alert();
+                    error!(*log_context, "[diem_vm] UNREACHABLE");
+                    return Ok(discard_error_vm_status(VMStatus::Error(
+                        StatusCode::UNREACHABLE,
+                    )));
                 }
-            }
-            TransactionPayload::Module(_) | TransactionPayload::Script(_) => {
-                log_context.alert();
-                error!(*log_context, "[diem_vm] UNREACHABLE");
-                return Ok(discard_error_vm_status(VMStatus::Error(
-                    StatusCode::UNREACHABLE,
-                )));
-            }
+            },
+            TransactionMetadata::new(&txn),
```

Comment:
It looks like the branch does not have #6891 so you should be able to use the txn_data that was created at the beginning of process_writeset_transaction(). Either that, or you should delete the unused txn_data instance.

Reply:
Fixed this!